keps/sig-network/4410-k8s-network-interface/README.md

```diff
@@ -0,0 +1,113 @@
+- Handle support levels like Gateway API (e.g. "core" and "extended")
+- Handle implementation-specific use cases through extension points
+- Decouple the Pod and Node Network setup
+- Provide garbage collection to ensure no resources created during pod setup such as Linux bridges, ebpf programs, 
+allocated IP addresses are left behind after pod deletion
+- Improve the current IP handling for pods (PodIP) to be handle multiple IP addresses and 
+a field to identify the IP address family (IPV4 vs IPV6)
+- Provide backwards compatibility for the existing CNI approach and migration a path to fully adopt KNI
+- Guarantee the network is setup and in a healthy state before containers are started (ephemeral, init, regular)
+- If feasible, provide API awareness of Pod network namespaces (e.g. interface names)
+- Provide a uniform approach for network setup/teardown for both virtualized (kata) and non-virtualized (runc) 
+runtimes including kubevirt. This could eliminate the high and low level runtimes from the networking path
+- Provide a reference implementation of the KNI network runtime
+- Provide the ability to have all the dependencies packaged in the container image (no more CNI binaries in the host file system)
+..- No more downloading CNI binaries via initContainers/Mounting /etc/cni/net.d or /opt/cni/bin
+- Provide the ability to use native k8s resources for configuration such as a ConfigMap's instead of configuration files in host file system
+- Provide an API to indicate network readiness for the node (no more files on disk)
+- Eliminate the need to exec binaries and replace with gRPC
+- Make troubleshooting easier by having logs accessible via kubectl logs
+- Improve network pod startup time
+- Provide the ability to prevent additional scheduling of pods if IPAM is out of IP addresses without evicting running pods
+
+### Non-Goals
+
+1. Any changes to the kube-scheduler 
+2. Any specific implementation other than the reference implementation. However we should ensure the KNI-API is flexible enough to support
+
+## Proposal
+
+The proposal of this KEP is to design and implement the KNI-API and make necessary changes to the CRI-API and container runtimes. The scope should be kept to a minimum and we should target feature parity. 
+
+### User Stories
+
+We are constantly adding these user stories, please join the community sync to discuss. 
+
+#### Story 1
+
+As a cluster operator, I need the ability to determine my network(s) is ready so that my pods come up with a working network.
+
+#### Story 2
+
+As a cluster operator, I need the ability to determine what networks are available on my node so that upstream components can ensure the pod is scheduled on the appropriate node. 
+
+#### Story 3
+
+As a Kubernetes developer, I need the ability to have extension points for pod network setup, teardown and update so that I can support future Kubernetes networking features with either reducing the changes to core kubernetes or eliminating them
```

Comment:
So, I'm reading this as not just setup and teardown, but, within a pod's lifecycle, maybe we can expand this to say that it's not just "update" but during it's lifecycle, can that be more explicit about taking actions while the pod is running?

> I need the ability to have extension points for pod network setup, teardown and updates while pods are running

I think of this as a current limitation at having execution points on pod creation or deletion. 

Also this could be split into its own thing, because there's a kinda double thing here with both extension points and reducing or eliminating changes to core.

Reply:
" kubernetes developer that wants to do network things " is not an user story, you need to define the user story that require those advanced networking features , and then we figure out what is thebest solution